adapters/coap/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java

```diff
@@ -70,11 +76,18 @@ private CoapContext(
             final DeviceUser originDevice,
             final DeviceUser authenticatedDevice,
             final String authId,
+            final boolean isResponseTimestampEnabled,
             final Span span) {
         super(span, authenticatedDevice);
         this.exchange = exchange;
         this.originDevice = originDevice;
         this.authId = authId;
+        this.isResponseTimestampEnabled = isResponseTimestampEnabled;
+    }
+
+    private static boolean shouldResponseIncludeTimeOption(final CoapExchange request) {
```

Comment:
FMPOV we should turn this into a non-static method and perform the check on the class' *exchange* field when needed.